website/blog/2024-10-28-kustomize-vs-helm/index.mdx

```diff
@@ -0,0 +1,179 @@
+
+Unlike Helm, which relies on templates, Kustomize uses overlays to apply different configurations to a base set of resources. This allows you to maintain base YAML files and layer environment-specific changes, such as for `development`, `staging`, or `production`.
+
+> We recently published an [in-depth Kustomize tutorial](/blog/patching-with-kustomize/) that explores the various types of patches, transformers, and generators, along with best practices for using the tool effectively. Be sure to check it out for a more detailed guide.
+
+### How to use Kustomize?
+
+Kustomize allows you to manage and customize Kubernetes configurations by layering modifications without altering the original files. A typical Kustomize setup involves a directory structure consisting of a base and overlays.
+
+![How to use Kustomize](https://github.com/user-attachments/assets/1844cdcb-507d-43e6-b5f1-8daaef4e9dd0)
+
+### Kustomise project structure
+
+The `base/` directory contains a [kustomization.yaml](/blog/patching-with-kustomize/#the-role-of-the-kustomizeyaml-file) file along with manifest files for common resources shared across environments. The `kustomization.yaml` file defines the core resources, like `Deployments` and `Services`, that apply to all environments.
+
+The `overlays/` directories contain their own `kustomization.yaml` files, which reference the resources from the base and apply environment-specific patches or customizations. These patches could include changes to image tags, resource limits, labels, or other environment-specific configurations. The overlay directories allow you to apply modifications for environments such as development, staging, or production, without altering the core configurations in the base.
+
+```
+├── base/
+│   ├── kustomization.yaml
+│   ├── deployment.yaml
+│   ├── service.yaml
+└── overlays/
+    ├── production/
+    │   ├── kustomization.yaml
+    │   └── replica-patch.yaml 
+    └── staging/
+        ├── kustomization.yaml
+        └── resource-patch.yaml
+```
+
+## Kustomize vs Helm, what are the differences?
+
+| **Feature**                   | **Kustomize**                                      | **Helm**                                                             |
+|--------------------------------|---------------------------------------------------|----------------------------------------------------------------------|
+| **Native Kubernetes integration** | Yes (since v1.14)                                 | No                                                                   |
```

Comment:
I don't think I know what you mean by native kubernetes integration. Do you mean `kubectl` integration?

Reply:
Yeah! I mean it is native to Kubernetes through Kubectl, but I get the confusion. I'll just use "kubectl integration" instead